corehq/apps/userreports/sql/adapter.py

```diff
@@ -44,30 +45,40 @@ def get_query_object(self):
         """
         return self.session_helper.Session.query(self.get_table())
 
+    def best_effort_save(self, doc):
+        """
+        Does a best-effort save of the document. Will fail silently if the save is not successful.
+
+        For certain known, expected errors this will do no additional logging.
+        For unexpected errors it will log them.
+        """
+        try:
+            self.save(doc)
```

Comment:
Oh, I see this is still all or nothing in terms of the rows that get saved. So if you have three rows [good, bad, good], none of them will be saved because of the one bad row. Is that what you intended? Or did you want to save the two good rows and discard the bad one?